### ISSUE TITLE
Rules are not using "warn" severity for certain issue types

### ISSUE THREAD

**Issue by shawnmcknight** [webpro-nl/knip#713]
I've created a reproduction for this issue at https://github.com/shawnmcknight/knip-warn.

I'm finding that certain issue types are not respecting the "warn" configuration in the `rules` section of the `knip.json` file. Specifically, I am seeing that "files", and "Unused devDependencies" are showing as errors even though the configuration has them set to "warn".

In the reproduction I create a simple repo which has some unused dependencies (`express`), unused dev dependencies (`@faker-js/faker`), an unused file (`orange.ts`), and an unused export (`unusedApple` in `apple.ts`). My `knip.json` file is:
```json
{
  "$schema": "https://unpkg.com/knip@5/schema-jsonc.json",
  "rules": {
    "files": "warn",
    "dependencies": "warn",
    "unlisted": "warn",
    "unresolved": "warn",
    "exports": "warn",
    "types": "warn",
    "enumMembers": "off"
  }
}
```

In this scenario, I would expect every issue to be reported as a warning, but both "Unused files" and "Unused devDependencies" are being reported as errors:
![image](https://github.com/webpro-nl/knip/assets/4583869/11be3ffe-3c46-44de-95b7-6db2307e624c)

The "Unused dependencies" and "Unused exports" are both reporting as warnings as expected.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.24.1. See [Release 5.24.1](https://github.com/webpro-nl/knip/releases/tag/5.24.1) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

**Comment by webpro** [comment]
Thanks @shawnmcknight! This area received little attention for some time, should be better now.

**Comment by shawnmcknight** [comment]
@webpro Thanks for looking into this so quickly.

I've tested again with the latest version and the "Unused files" are now correctly reporting at "warn" level, but the "Unused devDependencies" are still triggering at "error" level.

After taking a peek at your PR, it appeared to me that the `rules` configuration actually separates different dependency types from each other. I tried modifying my `knip.json` file to have a separate `devDependencies` property with a `warn` level and when this was set, the output correctly reflected warnings for the unused dev dependencies.

I think this might simply now be a json schema and documentation issue, which I could log separately if that's helpful. The json schema for `rules` appears to be missing `devDependencies`, `optionalPeerDependencies`, and `binaries`. The [documentation on issue types](https://knip.dev/reference/issue-types) has "Unused devDependencies" and "Referenced optional peerDependencies" all grouped under the `dependencies` key but it seems like those should have separate keys based on the zod schema.

If my interpretation here is correct, I'm happy to submit a PR to update the docs and json schema.

Thanks!
